pkgs/development/libraries/libvirt/default.nix

```diff
@@ -16,6 +16,19 @@ with stdenv.lib;
 # if you update, also bump <nixpkgs/pkgs/development/python-modules/libvirt/default.nix> and SysVirt in <nixpkgs/pkgs/top-level/perl-packages.nix>
 let
   buildFromTarball = stdenv.isDarwin;
+  # libvirt hardcodes the binary name 'ebtables', but in nixpkgs the ebtables
+  # binary we want to use is named 'ebtables-legacy'.
+  # Create a derivation to alias the binary name so that libvirt can find the right one, and use that below.
+  ebtables-compat = stdenv.mkDerivation {
+    pname = "ebtables-compat";
+    version = ebtables.version;
+    src = null;
+    buildInputs = [ ebtables ];
+    buildCommand = ''
+      mkdir -p $out/bin
+      ln -sf ${ebtables}/bin/ebtables-legacy $out/bin/ebtables
```

Comment:
Can we move this change into the normal ebtables?

Reply:
If we rename it for everyone, I think it'll lead to conflicts elsewhere. Specifically, there's a binary named 'ebtables' in 'iptables-nftables-compat' already, which is what this should eventually switch to use (once the nixos default firewall switches to iptables-nftables I think?).

I also suspect renaming the ebtables-legacy binary would require updating quite a few packages. I'm willing to do it if we think it's the right path, but I'm not convinced it is.

One other alternative could be to patch libvirt's build system to allow configuring this path, which the autotools libvirt builder supported (https://github.com/euank/nixpkgs/blob/6485019f6aaf6d23dbd5699f38d5e947f6504c2a/pkgs/development/libraries/libvirt/default.nix#L66), but the current meson builder doesn't, resulting in needing a hack like this or some other solution.

Do you have any thoughts on what the right path is here given the above? Is there anything else I should be considering?

Reply:
If we might break other packages we shouldn't rename it for all of them.
If you don't want to redo the buildsystem than I think your current approach is totally fine.